packages/amesos2/example/GappedMtxGIDs-1proc.cpp

```diff
@@ -421,9 +424,24 @@ readCrsMatrixFromFile (const std::string& matrixFilename,
           gblRowInd -= 1 ;
           gblColInd -= 1 ;
         }
-        A->insertGlobalValues (gblRowInd, LO (1), &val, &gblColInd);
+        counts[gblRowInd]++;
+        vals.push_back(val);
+        gblRowInds.push_back(gblRowInd);
+        gblColInds.push_back(gblColInd);
       }
     }
+
+    // Max number of entries in any row
+    using pair_type = decltype(counts)::value_type;
+    auto pr = std::max_element(
+        std::begin(counts),
+        std::end(counts),
+        [] (pair_type const& p1, pair_type const& p2){ return p1.second < p2.second; }
+    );
+    A = Teuchos::rcp(new MAT(rowMap, pr->second));
```

Comment:
Be careful -- if `counts` is empty, `pr` is invalid.  Suggested change:
```c++
const size_t maxCount = counts.empty () ? size_t (0) : pr->second;
A = Teuchos::rcp (new MAT (rowMap, maxCount));
```

Reply:
Thanks @mhoemmen!  The last push implements your suggestion.